app/containers/ITunesSearch/saga.js

```diff
@@ -0,0 +1,21 @@
+import { takeLatest, put, call } from 'redux-saga/effects';
+import { iTunesSearchTypes, iTunesSearchCreators } from './reducer';
+// Individual exports for testing
+import { getSongs } from '@services/iTunesApi';
+const { REQUEST_GET_ITUNES_SONGS } = iTunesSearchTypes;
+const { successGetItunesSongs, failureGetItunesSongs } = iTunesSearchCreators;
+
+export function* getItunesSongs(action) {
+  const response = yield call(getSongs, action.songName);
+
+  const { data, ok } = response;
+  if (ok) {
+    yield put(successGetItunesSongs(data));
+  } else {
+    yield put(failureGetItunesSongs(data));
```

Comment:
add the ternary operation for getting the error here, and then send it to the reducer